language/bytecode-verifier/src/dependencies.rs

```diff
@@ -210,7 +210,11 @@ fn verify_imported_structs(context: &Context) -> PartialVMResult<()> {
             Some(def_idx) => {
                 let def_handle = owner_module.struct_handle_at(*def_idx);
                 if !compatible_struct_abilities(struct_handle.abilities, def_handle.abilities)
-                    || !compatible_type_parameters(
+                    || !compatible_type_parameter_constraints(
+                        struct_handle.type_param_constraints(),
+                        def_handle.type_param_constraints(),
+                    )
+                    || !compatible_type_parameters_phantom_decl(
```

Comment:
I would probably keep this as one function called `compatible_struct_type_parameters` which checks the phantom  property, and then the ability one